packages/api/src/external/carequality/document/__tests__/create-outbound-document-retrieval-req.test.ts

```diff
@@ -164,3 +175,95 @@ describe("outboundDocumentRetrievalRequest", () => {
+    const outboundDocumentQueryResps: OutboundDocumentQueryResp[] = [
+      makeOutboundDocumentQueryResp({
+        gateway: makeXcaGateway({ homeCommunityId: homeCommunityId1 }),
+        documentReference: [
+          makeDocumentReferenceWithMetriportId({ homeCommunityId: homeCommunityId1 }),
+          makeDocumentReferenceWithMetriportId({ homeCommunityId: homeCommunityId1 }),
+        ],
+      }),
+      makeOutboundDocumentQueryResp({
+        gateway: makeXcaGateway({ homeCommunityId: homeCommunityId2 }),
+        documentReference: [
+          makeDocumentReferenceWithMetriportId({ homeCommunityId: homeCommunityId2 }),
+          makeDocumentReferenceWithMetriportId({ homeCommunityId: homeCommunityId2 }),
+        ],
+      }),
+    ];
+
+    const res: OutboundDocumentRetrievalReq[] = createOutboundDocumentRetrievalReqs({
+      patient,
+      requestId,
+      initiator,
+      outboundDocumentQueryResps,
+    });
+
+    expect(res).toBeTruthy();
+    expect(res.length).toEqual(2);
+
+    expect(res[0]?.documentReference?.length).toEqual(2);
+    expect(res[0]?.documentReference).toEqual(outboundDocumentQueryResps[0]?.documentReference);
+
+    expect(res[1]?.documentReference?.length).toEqual(2);
+    expect(res[1]?.documentReference).toEqual(outboundDocumentQueryResps[1]?.documentReference);
+  });
+  it("correct responses with multiple outboundDocumentQueryResps, where the doc refs for one gateway have different home community ids", async () => {
+    const homeCommunityId1 = faker.string.uuid();
+    const homeCommunityId2 = faker.string.uuid();
+    const outboundDocumentQueryResps: OutboundDocumentQueryResp[] = [
+      makeOutboundDocumentQueryResp({
+        gateway: makeXcaGateway({ homeCommunityId: homeCommunityId1 }),
+        documentReference: [
+          makeDocumentReferenceWithMetriportId({ homeCommunityId: homeCommunityId1 }),
+          makeDocumentReferenceWithMetriportId({ homeCommunityId: homeCommunityId1 }),
+        ],
+      }),
+      makeOutboundDocumentQueryResp({
+        gateway: makeXcaGateway({ homeCommunityId: homeCommunityId2 }),
+        documentReference: [
+          makeDocumentReferenceWithMetriportId({ homeCommunityId: homeCommunityId2 }),
+          makeDocumentReferenceWithMetriportId({ homeCommunityId: homeCommunityId2 }),
+        ],
+      }),
+    ];
+
+    const res: OutboundDocumentRetrievalReq[] = createOutboundDocumentRetrievalReqs({
+      patient,
+      requestId,
+      initiator,
+      outboundDocumentQueryResps,
+    });
+
+    expect(res).toBeTruthy();
+    expect(res.length).toEqual(2);
+
+    expect(res[0]?.documentReference?.length).toEqual(2);
+    expect(res[0]?.documentReference).toEqual(outboundDocumentQueryResps[0]?.documentReference);
+
+    expect(res[1]?.documentReference?.length).toEqual(2);
+    expect(res[1]?.documentReference).toEqual(outboundDocumentQueryResps[1]?.documentReference);
+  });
+});
+
+describe("filterDocRefsWithMetriportId", () => {
```

Comment:
This should be placed on the respective test file, `packages/api/src/external/carequality/document/__tests__/shared.test.ts`